### DISCUSSION TITLE
Support: How to improve the related posts

### DISCUSSION THREAD

**Discussion by lsolesen** [mmistakes/minimal-mistakes#2545] [Q&A] [answered]
I am using Github Actions to publish my site and Jekyll 4. Is there a way to improve the way the related posts are shown - and not just the most current like it works now?

**Comment by iBug** [comment]
Add this to your `Gemfile`:

```ruby
gem "classifier-reborn"
```

Add this to your `_config.yml`:

```yaml
lsi: true
```

Trust me, it's *very* slow and increased build time by over a minute on my development machine (i7-8850H, Ubuntu 19.10).

**Comment by iBug** [comment, marked as answer]
[This article](https://frankindev.com/2019/11/21/enable-related-posts-with-lsi/) describes the use of `gsl` (GNU Scientific Library) to speed up LSI. I just tried it out and it's so ridiculously fast! Build time was shortened from 70+ seconds to just 3s on my local machine, and from almost 3 minutes to just 8s on GitHub Actions.

**Comment by lsolesen** [comment]
Cool. Thanks. Is there a workaround to get related to work properly in
collections?

**Comment by mmistakes** [comment]
The object is `related_posts` e.g. it's going to give you related posts. If you want "related collections" that's plugin territory I believe. Don't think that comes with Jekyll natively like `site.related_posts`

**Comment by lsolesen** [comment]
@mmistakes Yes, I know. Was just a related question, as any of you have worked with collections as regards to related posts, tags, categories and so on yet for Jekyll. But you can close this, if you like. Thanks @iBug.

**Comment by lsolesen** [comment]
@iBug Yeah, tried to build on Github Actions, but it is really struggling after setting --lsi, even though I thought I installed GSL, but have to investigate a little furter: https://github.com/lsolesen/legestue-net-jekyll/blob/master/.github/actions/build-dist-site/Dockerfile and https://github.com/lsolesen/ifire-dk-jekyll/runs/713224422

**Comment by iBug** [comment]
@lsolesen Did you add `gem "gsl"` to your `Gemfile`?
